src/main/java/com/artipie/asto/fs/RxFile.java

```diff
@@ -160,20 +165,6 @@ public Completable delete() {
      * @return File size in bytes.
      */
     public Single<Long> size() {
-        return Single.defer(
-            () -> {
-                final SingleSubject<Long> res = SingleSubject.create();
-                this.exec.submit(
-                    () -> {
-                        try {
-                            res.onSuccess(Files.size(this.file));
-                        } catch (final IOException iex) {
-                            res.onError(iex);
-                        }
-                    }
-                );
-                return res;
-            }
-        );
+        return Single.fromCallable(() -> Files.size(this.file));
```

Comment:
@Sammers21 that is a blocking operation executing in common thread pool. Before the changes the blocking operation was performed on special thread pool

Reply:
RxJava can also be reconfigured to use the Vert.x scheduler:

```
RxJavaPlugins.setComputationSchedulerHandler(s -> RxHelper.scheduler(vertx));
RxJavaPlugins.setIoSchedulerHandler(s -> RxHelper.blockingScheduler(vertx));
RxJavaPlugins.setNewThreadSchedulerHandler(s -> RxHelper.scheduler(vertx));
```